src/semidiscretization/semidiscretization.jl

```diff
@@ -15,6 +15,23 @@ Return the number of degrees of freedom associated with each scalar variable.
     ndofs(mesh, solver, cache)
 end
 
+"""
+    ndofsglobal(semi::AbstractSemidiscretization)
+
+Return the global number of degrees of freedom associated with each scalar variable.
+This is the same as [`ndofs`](@ref) for simulations running in serial or
+parallelized via threads. It will in general be different for simulations
+running in parallel with MPI.
+"""
+@inline function ndofsglobal(semi::AbstractSemidiscretization)
+    mesh, _, solver, cache = mesh_equations_solver_cache(semi)
+    ndofsglobal(mesh, solver, cache)
+end
+
+@inline function ndofsglobal(mesh, solver, cache)
+    ndofs(mesh, solver, cache)
+end
```

Comment:
Is this fallback required? We have

https://github.com/trixi-framework/Trixi.jl/blob/01032aaa78f6140d7200f50d1011b3c7cd0c9c9f/src/solvers/dg.jl#L460-L462

which should do the job, doesn't it?

Reply:
`FDSBP <: DG`

Reply:
Ah! Thanks for that!

I now removed the fallback, i.e. only `DGMulti` should error when run with MPI.

Reply:
What do you think about this, @sloede?

Reply:
Think about what exactly (sorry, I missed this before)?

Reply:
The full PR - and the handling of special callbacks and error messages with MPI in particular

Reply:
I think I am ok with the approach as it is. We might need to revise once we have gained some practical experience, but I think we should just move forward here now.